outputconsumer.go

```diff
@@ -101,5 +103,11 @@ func (e *outputConsumer) wait() error {
 		e.mutex.Lock()
 		e.opts.Logger.Println("Encountered timeout")
 		return fmt.Errorf("after %s: %w", e.opts.Timeout, TimeoutError)
+	case state := <-ttExited(e.tt, true):
```

Comment:
I guess this is an error condition, because normally the other selects would have fired before this one? Please add a comment explaining.

Though I wonder if this really should be here. Processes exiting non-zero is not an error condition to termtest. Termtest is meant to be used for testing processes in a terminal context, and failure conditions such as non-zero exits are part of that scope. Again though, I might be missing something, so if we want to keep this definitely add a comment.

Reply:
Not quite. The first case would have fired if there is an output match. The second case would have fired if an ultimate timeout was reached, and the process was still active. The third case (this one) will fire if the process exists (either normally, or abnormally; it does not matter) without either of the two cases having (yet) fired.

The most common case of this firing is if we typo an expectation and the process exits normally with unmatched output. I assume this is exactly the case we're trying to catch with this ticket.

Reply:
Nice, so this would basically alert you that you forgot a `tt.ExpectExit()` of some sort? That's a nice win!

Reply:
In some cases, yes, I suppose so.